mlflow/pipelines/steps/train.py

```diff
@@ -414,6 +414,7 @@ def _resolve_estimator_plugin(self, plugin_str, X_train, y_train):
             f"{estimator.__class__.__module__}.{estimator.__class__.__name__}"
         )
         self.best_parameters = best_parameters
+        self._write_best_parameters_outputs(best_parameters, {}, output_directory)
```

Comment:
I think we will need to change the behavior in `_write_best_parameters_outputs`.

Currently it will output in this form
```
# tuned hyperparameters
alpha: 0.1

# hardcoded parameters

```

but that doesn't make sense in the context of automl.

Reply:
Do we need to write an empty "# hardcoded parameters" bucket for tuning runs?  I will check if the respective dict has any values and write the header only in this case. Hope this is ok for you.

Reply:
Done!

Reply:
sorry my comment was unclear. the issue is that automl parameters will be written under the heading "# tuned hyperparameters", but that is not accurate. instead, we should extend `_write_best_parameters_outputs` to accept a automl parameters argument, and then output those under the heading "# automl parameters"

Reply:
the empty dict for hardcoded params is not a problem IMO

Reply:
ok, pushed this one as well!